guides/puppet-web-apps/html-views/README.md

```diff
@@ -0,0 +1,98 @@
+
+## Partial View-Model
+
+A partial view-model is a valid JSON document with the filename extension `.json` that contains an `Html` property that points to the corresponding partial HTML view. It will also contain properties for the data that will be bound between the view and the database model.
+
+The partial view-model can be combined with an optional view-model code-behind which defines the class name and contains input handlers. The filename extension for this file is `.json.cs`.
+
+A partial HTML view together with its corresponding partial view-model can simply be referred to as a partial.
+
+### Using Partials
+
+A partial can be accessed in two different ways:
+
+1. Directly through an HTTP request that is handled in the application by a `Handle.GET`. The response to this request should be a complete HTML and JSON document. 
+2. Through an **insertion point**. An insertion point is a combination of a `Self.GET` call, which acts as the partial view-model insertion point, and a [`<starcounter-include>`](https://github.com/Polyjuice/starcounter-include) tag, which acts as a partial HTML view insertion point. The `<starcounter-include>` tag determines where in the DOM tree the partial HTML view should be rendered.
+
+#### Examples
+
+The examples use [Web Component](/guides/puppet-web-apps/introduction-to-web-components) features and work best in Google Chrome. Other browsers are supported using [polyfills](http://webcomponents.org/polyfills/).
+
+##### Creating the Handler
+
+No matter if the partial is accessed through an HTTP request from the browser or a `Self.GET` call, there always needs to be a handler that deals with the request. This handler simply returns a partial containing the JSON and HTML. The handler, in its simplest form, looks something like this:
+
+```cs
+Handle.GET("/your/partial/url", () => 
+{
+    return new YourPartialPage();
+}
+```
+
+Keep in mind that this only returns the JSON and HTML if the app is using the `HtmlFromJsonProvider` and `PartialToStandaloneHtmlProvider` [middlerware](/guides/network/middleware/).
+
+##### Adding the Insertion Point
+
+It is now possible to create an insertion point for this partial. The way to do this is by attaching the partial to a parent partial using `Self.GET`. For example:
+
+```cs
+mainPage.SubPage = Self.GET("/your/partial/url");
+```
+
+In the partial HTML view for the `mainPage` above, the HTML from the `SubPage` partial can be stamped in like so:
+
+```html
+<starcounter-include partial="{{model.SubPage}}"></starcounter-include>
+```
+
+##### Partial HTML View Example
+
+A partial HTML view may look something like this:
+
+{% raw %}
+```html
+<!--
+    Load you dependencies: <script>s, HTML Imports, CSS stylesheets etc.
+    Those dependencies will only be executed once when the partial is imported
+-->
+<!-- For example, to use Polymer's dom-bind custom element: -->
+<link rel="import" href="/sys/polymer/polymer.html" />
+<style>
+.myapp-address-entry__name {
```

Comment:
This is not correct BEM.  For a correct BEM, it should be `.myapp-address-entry-name`

In BEM, everything at the root level of a view is Block. A Block can consist of multiple Elements (prefixed with `__`). Blocks and Elements can have Modifiers (prefixed with`--`).

Should the above sentence be added to https://docs.starcounter.io/guides/mapping-and-blending/avoiding-css-conflicts/ for clarification?